### ISSUE TITLE
Trying to reset size of table as Tabulator does not reset based on container size 

### ISSUE THREAD

**Issue by anonthemon** [olifolkerd/tabulator#2078]
I am using the latest Tabulator version and I am trying to get the table to resize based on the container resizing when a slider navigation bar is hidden and moved to the left so the table automatically increases its width. However, I have tried to do table.redraw() when my slider is moved. That does not work. I have tried something like this:

`document.getElementsByClassName("tabulator").style.width = "1500px";`

But this is not working either. Is there a way I can manually change the width when I'm clicking my slider since it's not resizing with the increase in its container width? I can set the width initially based on the class with this:

```
.tabulator{
    width:1280px;
}
```

But I cannot figure out how to do it dynamically.

**Comment by anonthemon** [comment]
The width will increase but then the rest of the table is blank so the columns aren't expanding to be the size of the new width. This is when I do this:

`var elements = document.getElementsByClassName("tabulator");
    for(var i = 0, length = elements.length; i < length; i++) {
       
          elements[i].style.width = '1500px';
       
    }`

**Comment by anonthemon** [comment]
However, when I resize the window the table then automatically readjusts and fills in the blank space. Am I doing something wrong with calling the redraw function? It isn't working.

**Comment by olifolkerd** [comment]
Hey @anonthemon 

This issue has been covered exteeeensivly in the documentation and in probably 100 issues on both here and stack overflow.

No element in JavaScript is aware of when it is resized, aside from the window. therfore if the table is resized you MUST call the ***redraw*** function on the table in order for it to adjust to display the correct information.

ES8 introduced the ResizeObserver which allows elements to know when they have been resized, and tabulator uses this feature where available, which is why on Chrome, tabulator does automatically resize, but until the other browsers adopt this you will need to handle that part yourself as outlined above.

I hope that helps,

Cheers

Oli :)

**Comment by michael-ts** [comment]
@olifolkerd I would say the documentation is lacking if there have been 100 issues with people using the feature. :-D

I'm using Chrome, and I do NOT see the behavior you mention.  I tried setting `responsiveLayout` to "collapse" and then "hide" and there was apparently some kind of bad interaction with `persistentLayout` because I have *not* been able to figure out how to restore hidden columns.  I'm resizing the Div containing the tabulator, and when the page reloads it "remembered" that the columns were hidden despite there being plenty of room.  I tried turning off `persistentLayout` and `responsiveLayout` in various combinations this partially fixed the issue, only now the whole table collapses down to some minimum size if `persistentLayout` is off, which I can resize but then it doesn't save, but if `persistentLayout` is on I only see my first column and can't figure out how to get any other columns back.  This is all with `persistentLayout` not even mentioned as a tag.

So now I have to waste a bunch of time trying to figure out if I did something wrong, or if there is a bug, or what... as well as try to restore the original layout.
